tchannel/peer_strategy.py

```diff
@@ -30,3 +32,24 @@ def get_score(self, peer):
 class ZeroScoreCalculator(ScoreCalculator):
     def get_score(self, peer):
         return 0
+
+
+class PreferIncomingCalculator(ScoreCalculator):
+    def get_score(self, peer):
+        """Calculate the peer score based on connections.
+
+        If the peer has no incoming connections, it will have largest score.
+        In our peer selection strategy, the largest number has least priority
+        in the heap.
+
+        If the peer has incoming connections, we will return number of outbound
+        pending requests and responses.
+
+        :param peer: instance of `tchannel.tornado.peer.Peer`
+        :return: score of the peer
+        """
+        if (peer.host == '0.0.0.0' or int(peer.port) == 0 or
+                not peer.incoming_connections):
```

Comment:
I'm all for incremental improvements but this strategy just has a very bad behavior for the outgoing connections case. I would drop the incoming connections requirement for now until we can figure out a smarter strategy there.